src/ts/components/subtitletogglebutton.ts

```diff
@@ -65,22 +65,22 @@ export class SubtitleToggleButton extends ToggleButton<ToggleButtonConfig> {
     }
 
     private initPlayerEvents = () => {
-        this.player.on(this.player.exports.PlayerEvent.SubtitleAdded, this.checkSubtitleAvailability);
-        this.player.on(this.player.exports.PlayerEvent.SubtitleRemoved, this.checkSubtitleAvailability);
-        this.player.on(this.player.exports.PlayerEvent.PeriodSwitch, this.checkSubtitleAvailability);
-        this.player.on(this.player.exports.PlayerEvent.SourceUnloaded, this.checkSubtitleAvailability);
+        this.player.on(this.player.exports.PlayerEvent.SubtitleAdded, this.onAvailableSubtitlesChanged);
+        this.player.on(this.player.exports.PlayerEvent.SubtitleRemoved, this.onAvailableSubtitlesChanged);
+        this.player.on(this.player.exports.PlayerEvent.PeriodSwitched, this.onAvailableSubtitlesChanged);
+        this.player.on(this.player.exports.PlayerEvent.SourceUnloaded, this.onAvailableSubtitlesChanged);
     }
 
-    private checkSubtitleAvailability = () => {
-        const storedSubtitle: StoredSubtitleLanguage = StorageUtils.getObject(DummyComponent.instance().prefixCss('subtitlelanguage'));
+    private onAvailableSubtitlesChanged = () => {
+        const storedSubtitle: StoredSubtitleLanguage = StorageUtils.getObject(prefixCss('subtitlelanguage'));
         const subtitleList = this.player.subtitles.list();
 
         // only shows the button when subtitles are existing
         subtitleList.length > 0 ? this.show() : this.hide();
 
         // if the stored subtitle is set active and available, that subtitle is enabled
-        let subtitleToActivate: SubtitleTrack;
-        if (storedSubtitle && storedSubtitle.active && (subtitleToActivate = subtitleList.find(subtitle => subtitle.lang === storedSubtitle.language))) {
+        let subtitleToActivate: SubtitleTrack = subtitleList.find(subtitle => subtitle.lang === storedSubtitle.language);
```

Comment:
```suggestion
        const subtitleToActivate: SubtitleTrack = subtitleList.find(subtitle => subtitle.lang === storedSubtitle?.language);
```
if storedSubtitle can be null, this is a potential `null.language` access